csrc/torch_memory_saver.cpp

```diff
@@ -208,22 +248,36 @@ class TorchMemorySaver {
                 continue;
             }
 
-            CUmemGenericAllocationHandle newAllocHandle;
-            CUDAUtils::cu_mem_create(&newAllocHandle, metadata.size, metadata.device);
+            if (metadata.state == AllocationState::PAUSED) {
+                found_paused_allocation = true;
+
+                CUmemGenericAllocationHandle newAllocHandle;
+                CUDAUtils::cu_mem_create(&newAllocHandle, metadata.size, metadata.device);
 
-            CURESULT_CHECK(cuMemMap((CUdeviceptr) ptr, metadata.size, 0, newAllocHandle, 0));
+                CURESULT_CHECK(cuMemMap((CUdeviceptr) ptr, metadata.size, 0, newAllocHandle, 0));
 
-            CUDAUtils::cu_mem_set_access(ptr, metadata.size, metadata.device);
+                CUDAUtils::cu_mem_set_access(ptr, metadata.size, metadata.device);
 
 #ifdef TMS_DEBUG_LOG
-            std::cout << "[torch_memory_saver.cpp] TorchMemorySaver.resume"
-                      << " ptr=" << ptr << " metadata.size=" << metadata.size << " (old)metadata.allocHandle="
-                      << metadata.allocHandle
-                      << " (new)newAllocHandle=" << newAllocHandle << " tag=" << metadata.tag << " filter_tag=" << tag
-                      << std::endl;
+                std::cout << "[torch_memory_saver.cpp] TorchMemorySaver.resume"
+                          << " ptr=" << ptr << " metadata.size=" << metadata.size << " metadata.allocHandle="
+                          << metadata.allocHandle << " (new)newAllocHandle=" << newAllocHandle << " tag=" << metadata.tag << " filter_tag=" << tag
+                          << std::endl;
 #endif
 
-            metadata.allocHandle = newAllocHandle;
+                metadata.allocHandle = newAllocHandle;
+                metadata.state = AllocationState::ACTIVE;
+            }
+        }
+
+        if (!found_paused_allocation) {
+            std::string error_msg = "No paused allocations found";
+            if (!tag.empty()) {
+                error_msg += " for tag: " + tag;
+            }
+            error_msg += ". Cannot resume unpaused or non-existent allocations.";
+            std::cerr << "[torch_memory_saver.cpp] ERROR: " << error_msg << std::endl;
```

Comment:
nit: what about one line

```
std::cerr << .... << tag << ...
```

no worries, even if tag is empty, we can still print that tag